### PR TITLE
Managed Services Swagger Consistency Fixes

### PR DESCRIPTION
**Pull request by skayani**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2018-06-01-preview/managedservices.json [hunk 1]
```diff
@@ -759,9 +759,21 @@
...
+        },
         "roleDefinitionId": {
           "type": "string",
           "description": "The role definition identifier. This role will define all the permissions that the security group/service principal/user must have on the projected subscription. This role cannot be an owner role."
+        },
+        "delegatedRoleDefinitionIds": {
+          "type": "array",
+          "description": "The delegatedRoleDefinitionIds field is required when the roleDefinitionId refers to the User Access Administrator Role. It is the list of role definition ids which define all the permissions that the user in the authorization can assign to other security groups/service principals/users.",
+          "items": {
+            "format": "uuid",
```
Comment (dramuy) [inline specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2018-06-01-preview/managedservices.json:774]: guid?

Comment (dramuy) [reply specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2018-06-01-preview/managedservices.json:774]: Ignore it, you unuiqe identifier

**Comment by skayani** [comment]
Looking through these automatically marked breaking changes, they aren't actually breaking changes in the sense.  The additional delegatedRoleDefinitionIds, principalIdDisplayName, and details (in ErrorDefinition) should've been included in all API versions but was ultimately not.  However, none are required fields.

**Review by ChenTanyi — APPROVED** [review]

**Comment by skayani** [comment]
@ChenTanyi Thanks for approving this PR.  I see the automated comment mentioning next steps for breaking changes. As per my initial comment, do I need to take any additional steps given the fact that it isn't an actual breaking change PR (due to non-required fields being added)?

**Comment by ChenTanyi** [comment]
@skayani I don't see any difference from your initial comment. The property added is indeed breaking change. You need to go through the breaking change review to get breaking change approved.

**Comment by ChenTanyi** [comment]
If you have any question about it, please ask @jianyexi @akning-ms

**Comment by dramuy** [comment]
@ChenTanyi The properties added in this PR are already supported in the service and the swagger specs were not including those properties. As part of our fundamentals review; we received couple of alerts to fix the swagger correctness and completeness; which we are trying to address as part of this PR.

**Comment by ChenTanyi** [comment]
> @ChenTanyi The properties added in this PR are already supported in the service and the swagger specs were not including those properties. As part of our fundamentals review; we received couple of alerts to fix the swagger correctness and completeness; which we are trying to address as part of this PR.

@akning-ms

**Comment by akning-ms** [comment]
> @ChenTanyi The properties added in this PR are already supported in the service and the swagger specs were not including those properties. As part of our fundamentals review; we received couple of alerts to fix the swagger correctness and completeness; which we are trying to address as part of this PR.

Hi @dramuy, got your case, this can be a reason to get breaking change review board approval. You can refer https://aka.ms/AzBreakingChangesPolicy for detail process and fill a ADO work item refer below dashboard.
https://msazure.visualstudio.com/One/_dashboards/dashboard/f7db6895-7137-4695-b3f5-00dc4d36758f then send a mail or attend office hour which can be found in the policy document.

**Comment by skayani** [comment]
> > @ChenTanyi The properties added in this PR are already supported in the service and the swagger specs were not including those properties. As part of our fundamentals review; we received couple of alerts to fix the swagger correctness and completeness; which we are trying to address as part of this PR.
> 
> Hi @dramuy, got your case, this can be a reason to get breaking change review board approval. You can refer https://aka.ms/AzBreakingChangesPolicy for detail process and fill a ADO work item refer below dashboard.
> https://msazure.visualstudio.com/One/_dashboards/dashboard/f7db6895-7137-4695-b3f5-00dc4d36758f then send a mail or attend office hour which can be found in the policy document.
Hi @akning-ms I've taken a look at both the official guidelines you shared as well as the supplemental breaking changes guidelines here, and I'm still wondering about this being classified as a breaking change. As Ramu mentioned, the swagger specs from the very beginning should've included the properties principalIdDisplayName and delgatedRoleDefinitionIds but were not due to a mistake in the initial PR.  More importantly, the two fields are optional.  As for the refactor of the ErrorDefinition, this is provided in the server response when an error occurs and if you'll notice the only logical change made was to include an optional, read-only details property.  Please let me know your thoughts on the matter.

**Review by JeffreyRichter — APPROVED** [review]

**Comment by ChenTanyi** [comment]
@skayani Please let me know when the PR is ready to merge.

**Comment by skayani** [comment]
> @skayani Please let me know when the PR is ready to merge.

@ChenTanyi We're good to merge it now, thanks.

**Pull request merged**